### PR TITLE
fix use-after-free of Task with futures2

### PR DESCRIPTION
**Pull request by seanmonstar**

### CODE REVIEW

**Comment by carllerche** [comment]
cc @aturon

File: tokio-threadpool/src/lib.rs [hunk 1]
```diff
@@ -1090,7 +1092,6 @@ impl Inner {
         trace!("worker_terminated; num_workers={}", prev - 1);
 
         if 1 == prev {
-            trace!("notifying shutdown task");
```
Comment (seanmonstar) [inline tokio-threadpool/src/lib.rs]: Er, woops, did you want to keep this? I was just cleaning up a bunch of logs I had used to debug this, and thought this was one I had added.

**Comment by kpp** [comment]
@seanmonstar do you want me to rebase your PR onto new master?

**Comment by seanmonstar** [comment]
You can if you want! This has actually been waiting for @aturon to tell us if this breaks some other invariant I didn't consider.

**Comment by aturon** [comment]
@seanmonstar Sorry for the very delayed review! This PR LGTM.

**Comment by carllerche** [comment]
Closing due to inactivity and stagnation.

**Pull request closed without merging**